esp8266/modnetwork.c

```diff
@@ -109,6 +110,13 @@ STATIC mp_obj_t esp_connect(mp_uint_t n_args, const mp_obj_t *args) {
             p = "";
         }
         memcpy(config.password, p, len);
+        if (n_args > 3) {
+            bssid = mp_obj_str_get_data(args[3], &len);
+            config.bssid_set = 1;
+        } else {
+            bssid = "";
+        }
+        memcpy(config.bssid, bssid, len);
```

Comment:
I saw the unnecessary memcpy and overflow problem as well. :)
But I was primarily trying to follow the same coding style as the code that deals with ssid and password. Maybe I could make another PR later, that fixes these problems for all  these fields.